src/adagrams.js

```diff
@@ -1,15 +1,129 @@
+const letterPool = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+
+const letterScores = {
+  A: 1,
+  B: 3,
+  C: 3,
+  D: 2,
+  E: 1,
+  F: 4,
+  G: 2,
+  H: 4,
+  I: 1,
+  J: 8,
+  K: 5,
+  L: 1,
+  M: 3,
+  N: 1,
+  O: 1,
+  P: 3,
+  Q: 10,
+  R: 1,
+  S: 1,
+  T: 1,
+  U: 1,
+  V: 4,
+  W: 4,
+  X: 8,
+  Y: 4,
+  Z: 10,
+};
+
+const availbleLetters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+
 export const drawLetters = () => {
   // Implement this method for wave 1
+  let letterPoolCopy = JSON.parse(JSON.stringify(letterPool));
+  let letters = [];
+  for (let i = 0; i < 10; ) {
+    let singleLetter = availbleLetters.charAt(
+      Math.floor(Math.random() * availbleLetters.length)
+    );
+    if (
+      Object.values(letterPoolCopy).includes(singleLetter) > -1 &&
+      letterPoolCopy[singleLetter] > 0
+    ) {
+      letterPoolCopy[singleLetter] = letterPoolCopy[singleLetter] - 1;
+      letters.push(singleLetter);
+      i++;
+    }
+  }
+  return letters;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
   // Implement this method for wave 2
+  let bankLetters = Array.from(lettersInHand);
+  for (const letter of input) {
+    if (bankLetters.includes(letter)) {
+      bankLetters = bankLetters.filter((item) => item !== letter);
```

Comment:
There is a bug here that our tests don't catch (this may be causing the test failure in Learn).  `.filter` will check all of the letters in the letter bank, so if `letter` shows up multiple times, each instance will be removed.  If `input` has the same letter multiple times, even if the letter is in the letter bank the correct number of times, the first pass will find the letter, but the second pass will not.  For example, if the word is 'good' and the letter bank is ['g', 'o', 'o', 'd', 'x', 'x', 'x', 'x', 'x', 'x'], the function will return false when it gets to the second 'o' in the word.